apis/python/CMakeLists.txt

```diff
@@ -0,0 +1,107 @@
+#
+# The MIT License
+#
+# Copyright (c) 2024 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+cmake_minimum_required(VERSION 3.21)
+
+project(
+    ${SKBUILD_PROJECT_NAME}
+    VERSION ${SKBUILD_PROJECT_VERSION}
+    LANGUAGES CXX
+)
+
+find_package(pybind11 REQUIRED)
+
+find_library(
+    TILEDBVCF_LIB
+    NAMES tiledbvcf
+    PATHS "${CMAKE_CURRENT_SOURCE_DIR}/../../dist/lib"
+)
+
+if(NOT TILEDBVCF_LIB)
+    message(STATUS "Building libtiledbvcf from source")
+
+    file(MAKE_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/../../libtiledbvcf/build")
+
+    execute_process(
+        # Force Unix Makefiles to avoid issues with Ninja and htslib, which uses a custom build system
+        COMMAND bash -c "${CMAKE_COMMAND} -G 'Unix Makefiles' .. && make -j && make -j install-libtiledbvcf"
+        WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/../../libtiledbvcf/build"
+    )
+
+    find_library(
+        TILEDBVCF_LIB
+        NAMES tiledbvcf
+        PATHS "${CMAKE_CURRENT_SOURCE_DIR}/../../dist/lib"
+    )
+endif()
+
```

Comment:
Should probably check `TILEDBVCF_LIB-FOUND` here to make sure one of the statements succeeded.